### ISSUE TITLE
Problem with sparse arrays: missing values

### ISSUE THREAD

**Issue by bkamins** [JuliaSparse/SparseArrays.jl#15]
I am not sure where to report it, but I think there is a problem with handling `missing` values for sparse arrays:
```
julia> ms = sparse([1, 2], [1, 2], [10, missing])
2×2 SparseMatrixCSC{Union{Missing, Int64}, Int64} with 2 stored entries:
 10  ⋅
  ⋅   missing

julia> md = Matrix(ms)
2×2 Matrix{Union{Missing, Int64}}:
 10  0
  0   missing

julia> sparse(md)
ERROR: TypeError: non-boolean (Missing) used in boolean context

julia> ms[1, 2] = missing
ERROR: TypeError: non-boolean (Missing) used in boolean context

julia> ms[1, 1] = missing
missing

julia> ms
2×2 SparseMatrixCSC{Union{Missing, Int64}, Int64} with 2 stored entries:
  missing  ⋅
 ⋅          missing

julia> x = spzeros(3)
3-element SparseVector{Float64, Int64} with 0 stored entries

julia> y = similar(x, Union{Int, Missing})
3-element SparseVector{Union{Missing, Int64}, Int64} with 0 stored entries

julia> y[1] = missing
ERROR: TypeError: non-boolean (Missing) used in boolean context

julia> y[1] = 10
10

julia> y[1] = missing
missing
```

CC @nalimilan

**Comment by simonfxr** [comment]
I might have run into a similar problem with `Symbolics` `Num` type:
```julia
julia> using Symbolics, SparseArrays

julia> @variables x
1-element Vector{Num}:
 x

julia> sparse([x]) + sparse([x])
ERROR: TypeError: non-boolean (Num) used in boolean context
Stacktrace:
 [1] _binarymap_mode_1!(f::typeof(+), mx::Int64, my::Int64, xnzind::Vector{Int64}, xnzval::Vector{Num}, ynzind::Vector{Int64}, ynzval::Vector{Num}, rind::Vector{Int64}, rval::Vector{Num})
   @ SparseArrays /usr/lib/julia-18-git/share/julia/stdlib/v1.8/SparseArrays/src/sparsevector.jl:1305
 [2] _binarymap(f::typeof(+), x::SparseVector{Num, Int64}, y::SparseVector{Num, Int64}, mode::Int64)
   @ SparseArrays /usr/lib/julia-18-git/share/julia/stdlib/v1.8/SparseArrays/src/sparsevector.jl:1260
 [3] +(x::SparseVector{Num, Int64}, y::SparseVector{Num, Int64})
   @ SparseArrays /usr/lib/julia-18-git/share/julia/stdlib/v1.8/SparseArrays/src/sparsevector.jl:1384
 [4] top-level scope
   @ REPL[3]:1
```

The problem seems to be that symbolic equality does not return a boolean:
```julia
julia> typeof(zero(x) == x)
Num
```

But using `iszero(x)` on `Num` returns a boolean. So using `iszero(x) != true` would work for `Num` as well.

**Comment by SobhanMP** [comment]
@simonfxr this was fixed recently by #177

**Comment by simonfxr** [comment]
@SobhanMP Thats great! Works without a hitch on a fresh julia master build.
